### DISCUSSION TITLE
How to use EasyIDP ?

### DISCUSSION THREAD

**Discussion by Lyhour-Chhay** [UTokyo-FieldPhenomics-Lab/EasyIDP#8] [Q&A] [answered]
I really appreciate your kindness for sharing this valuable work. Currently, I am studying about 3D reconstruction of pavement surface texture. My purposes are to obtained the dense point cloud and depth image. I have been used Agisoft Metashape to generate it. I would like to tried your coding to generate the point cloud and 3D reconstruction. However, I am not sure how to run your coding. How can I input the images to generate the process ? Thank you very much.

**Comment by HowcanoeWang** [comment, marked as answer]
This tool is not designed for generating point clouds, you always need to generate point clouds and models by Agisoft. 

Instead, this tool is used to find (trace back) the same region from point cloud/3d models to raw images. Like this figure shows:
![image](https://user-images.githubusercontent.com/19968935/135056498-510ceb98-740e-410d-96b0-a7f1a1540f73.png)

**Reply by Lyhour-Chhay** [reply]
Thank you very much kindly response sir. Because I am not clearly understand your coding and purpose, I want to confirm that Does your coding generate the point cloud from the multi-view image ? Does your code conduct the SFM technique ? Thank you very much.

**Reply by HowcanoeWang** [reply]
> Does your coding generate the point cloud from the multi-view image

No, we only use the outputs of Agisoft, including points clouds and project XML files

> Does your code conduct the SFM technique

No, we use the SfM parameters after Agisoft reconstruction

**Reply by Lyhour-Chhay** [reply]
Thank you very much sir for your information. Anyway, do you know or have the coding Github which is generated the point clouds and depth image corresponding to RGB image ?

**Reply by HowcanoeWang** [reply]
Do you mean the open source SfM and MVS? Please check the following project, but hard to guarantee the ease-of-use and final product quality as commercial software:

* OpenMVG http://openmvg.readthedocs.io/en/latest/#
* libMV https://developer.blender.org/tag/libmv/
* VisualSFM http://ccwu.me/vsfm/
* Bundler http://www.cs.cornell.edu/~snavely/bundler/
* OpenDroneMap https://www.opendronemap.org/

And some other MVS open-source software packages
![image](https://user-images.githubusercontent.com/19968935/135080377-e4d23f34-37ea-4fa4-be54-5fd8935b8926.png)

**Reply by Lyhour-Chhay** [reply]
Thank you very much sir. I have been detail my problem statement via e-mail to you. I am looking forward to hearing from you soon. However, as you mentioned already, the final produce quality and the ease-of-use are difficult.
